stl/inc/algorithm

```diff
@@ -3520,11 +3520,22 @@ _CONSTEXPR20 pair<_RanIt, _RanIt> _Partition_by_median_guess_unchecked(_RanIt _F
 template <class _RanIt, class _Pr>
 _CONSTEXPR20_ICE void _Sort_unchecked(_RanIt _First, _RanIt _Last, _Iter_diff_t<_RanIt> _Ideal, _Pr _Pred) {
     // order [_First, _Last), using _Pred
-    _Iter_diff_t<_RanIt> _Count = 0;
-    while (_ISORT_MAX < (_Count = _Last - _First) && 0 < _Ideal) { // divide and conquer by quicksort
+    for (;;) {
+        if (_Last - _First <= _ISORT_MAX) { // small
+            _Insertion_sort_unchecked(_First, _Last, _Pred);
+            return;
+        }
+
+        if (_Ideal <= 0) { // heap sort if too many divisions

```

Comment:
I observe that `_Ideal` should never be negative. It's initially `_ULast - _UFirst` (non-negative) and then decreases via `_Ideal = (_Ideal >> 1) + (_Ideal >> 2)`. Perhaps an `_STL_INTERNAL_CHECK` for negative, and an exact `== 0` test otherwise? However, there's probably no codegen difference in practice, so I don't really care.

Reply:
Codegen is identical except for `je` vs. `jle` 
![image](https://user-images.githubusercontent.com/1544943/73115172-7cbeef00-3ed7-11ea-8940-8c3a4c4a19c9.png)